plugins/woocommerce/includes/class-wc-tracker.php

```diff
@@ -1169,4 +1178,239 @@ private static function get_mini_cart_info() {
+				WHERE order_id IN ($joined_ids)",
+				ARRAY_A
+			);
+			// phpcs:enable
+
+			foreach ( $data as $row ) {
+				$additional_data[ $row['order_id'] ] = array(
+					'woocommerce_version' => $row['woocommerce_version'],
+					'recorded_sales'      => $row['recorded_sales'] ? 'yes' : 'no',
+				);
+			}
+		} else {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			$data = $wpdb->get_results(
+				"SELECT post_id AS order_id, meta_key, meta_value
+				FROM $wpdb->postmeta
+				WHERE
+					meta_key IN (
+						'_order_currency',
+						'_order_total',
+						'_payment_method',
+						'_payment_method_title',
+						'_recorded_sales',
+						'_order_version'
+					)
+					AND post_id IN ($joined_ids)
+				",
+				ARRAY_A
+			);
+			// phpcs:enable
+
+			foreach ( $data as $row ) {
+				$meta_key = self::map_legacy_meta_key_name( $row['meta_key'] );
+				$additional_data[ $row['order_id'] ][ $meta_key ] = $row['meta_value'];
+			}
+		}
+
+		return $additional_data;
+	}
+
+	/**
+	 * Fetch refund data for a specific set of orders.
+	 *
+	 * @param array $order_ids List of order ID's to fetch data for.
+	 * @return array Refund data, indexed by order ID.
+	 */
+	private static function get_refund_order_data( $order_ids ) {
+		global $wpdb;
+
+		if ( empty( $order_ids ) || ! is_array( $order_ids ) ) {
+			return array();
+		}
+		$joined_ids  = implode( ',', $order_ids );
+		$refund_data = array();
+
+		if ( OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			$order_table_name = OrdersTableDataStore::get_orders_table_name();
+
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			$data = $wpdb->get_results(
+				"SELECT
+					parent_order_id AS order_id,
+					SUM(total_amount) AS refund_amount
+				FROM $order_table_name
+				WHERE
+					type = 'shop_order_refund'
+					AND status = 'wc-completed'
+					AND parent_order_id IN ($joined_ids)
+				GROUP BY parent_order_id",
+				ARRAY_A
+			);
+			// phpcs:enable
+		} else {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			$data = $wpdb->get_results(
+				"SELECT
+					refunds.post_parent AS order_id,
+					SUM(amount.meta_value) AS refund_amount
+				FROM $wpdb->posts AS refunds
+				LEFT JOIN $wpdb->postmeta AS amount ON amount.post_id = refunds.ID AND amount.meta_key = '_order_total'
+				WHERE
+					refunds.post_type = 'shop_order_refund'
+					AND refunds.post_status = 'wc-completed'
+					AND refunds.post_parent IN ($joined_ids)
+				GROUP BY refunds.post_parent",
+				ARRAY_A
+			);
+			// phpcs:enable
+		}
+
+		foreach ( $data as $row ) {
+			$refund_data[ $row['order_id'] ] = array(
+				'refund_amount' => $row['refund_amount'],
+			);
+		}
+
+		return $refund_data;
+	}
+
+	/**
+	 * Get a snapshot of the first 20 orders and the last 20 orders.
+	 *
+	 * @return array
+	 */
+	private static function get_order_snapshot() {
+		$first_20  = self::get_order_data( 'ASC', 20 );
+		$last_20   = self::get_order_data( 'DESC', 20 );
```

Comment:
'20' not necessary as is the default

Reply:
Same response [as above](https://github.com/woocommerce/woocommerce/pull/52074#discussion_r1806425845).